pkg/backend/display/events.go

```diff
@@ -151,14 +152,16 @@ func ConvertEngineEvent(e engine.Event, showSecrets bool) (apitype.EngineEvent,
 }
 
 func convertStepEventMetadata(md engine.StepEventMetadata, showSecrets bool) apitype.StepEventMetadata {
-	keys := make([]string, len(md.Keys))
+	keys := slice.Prealloc[string](len(md.Keys))
 	for i, v := range md.Keys {
 		keys[i] = string(v)
```

Comment:
This will panic. Assigning by index.
We might need a PreallocLength to accompany Prealloc.

Reply:
If this was a plain make before then nil-ness doesn't matter, I'll just change it back.